packages/x-date-pickers/src/internals/hooks/usePickerState.ts

```diff
@@ -69,99 +151,98 @@ export const usePickerState = <TInput, TDateValue>(
     }
   }, [parsedDateValue]);
 
-  const [draftState, dispatch] = React.useReducer(
-    (state: Draftable<TDateValue>, action: DraftAction<TDateValue>): Draftable<TDateValue> => {
-      switch (action.type) {
-        case 'reset':
-          return initDraftableDate(action.payload);
-        case 'update':
-          return {
-            ...state,
-            draft: action.payload,
-          };
-        default:
-          return state;
-      }
-    },
-    parsedDateValue,
-    initDraftableDate,
-  );
-  if (!valueManager.areValuesEqual(utils, draftState.committed, parsedDateValue)) {
-    dispatch({ type: 'reset', payload: parsedDateValue });
-  }
-
-  const [initialDate, setInitialDate] = React.useState<TDateValue>(draftState.committed);
-
-  // Mobile keyboard view is a special case.
-  // When it's open picker should work like closed, because we are just showing text field
-  const [isMobileKeyboardViewOpen, setMobileKeyboardViewOpen] = React.useState(false);
+  React.useEffect(() => {
+    if (isOpen) {
+      // Update all dates in state to equal the current prop value
+      setDate({ action: 'setAll', value: parsedDateValue, internal: true });
+    }
+  }, [isOpen]); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
It's rare when we need to use `// eslint-disable-line react-hooks/exhaustive-deps`. Why you added it?

Reply:
We don't want `parsedDateValue` as a deps, otherwise we would reset the state way to often